pilot/pkg/model/validation.go

```diff
@@ -1959,6 +1991,111 @@ func validateHTTPRewrite(rewrite *routingv2.HTTPRewrite) error {
 	return nil
 }
 
+// ValidateExternalService validates a external service.
+func ValidateExternalService(config proto.Message) (errs error) {
+	externalService, ok := config.(*routingv2.ExternalService)
+	if !ok {
+		return fmt.Errorf("cannot cast to external service")
+	}
+
+	if len(externalService.Hosts) == 0 {
+		errs = appendErrors(errs, fmt.Errorf("external service must have at least one host"))
+	}
+	for _, host := range externalService.Hosts {
+		errs = appendErrors(errs, validateHost(host))
+	}
+
+	servicePortNumbers := make(map[uint32]bool)
+	servicePorts := make(map[string]bool, len(externalService.Ports))
+	for _, port := range externalService.Ports {
+		if servicePorts[port.Name] {
+			errs = appendErrors(errs, fmt.Errorf("external service port name %q already defined", port.Name))
+		}
+		servicePorts[port.Name] = true
+		if servicePortNumbers[port.Number] {
+			errs = appendErrors(errs, fmt.Errorf("external service port %d already defined", port.Number))
+		}
+		servicePortNumbers[port.Number] = true
+	}
+
+	if externalService.Discovery == routingv2.ExternalService_NONE {
```

Comment:
Use a switch here with a `default` case that's an error so we don't get bitten when we add a new resolution type.